src/VisualStudio/Core/Def/SampleToolWindow/DocumentOutlineHelper.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    internal static class DocumentOutlineHelper
+    {
+        internal static List<DocSymbol> GetDocumentSymbols(DocumentSymbol[]? body)
+        {
+            var docSymbols = new List<DocSymbol>();
+            if (body is not null && body.Length > 0)
+            {
+                for (var i = 0; i < body.Length; i++)
+                {
+                    var documentSymbol = body[i];
+                    var ds = new DocSymbol(
+                        documentSymbol.Name,
+                        documentSymbol.Kind,
+                        documentSymbol.Range.Start.Line,
+                        documentSymbol.Range.Start.Character,
+                        documentSymbol.Range.End.Line,
+                        documentSymbol.Range.End.Character);
+                    var children = documentSymbol.Children;
+                    if (children is not null)
+                    {
+                        ds = AddNodes(ds, children);
+                    }
+
+                    docSymbols.Add(ds);
+                }
+            }
+
+            docSymbols = docSymbols.OrderBy(x => x.StartLine).ThenBy(x => x.StartChar).ToList();
```

Comment:
Can you explain the logic here? It appears that we are doing `OrderBy` to sort our list and then we call `Sort`. could we change `Sort` to accept an `IEnumerable<DocSymbol>` and just call `Sort` on `docSymbols`? or are the top level document symbols special in some way?

Reply:
We can change sort! Done